File: bb_waze_code/j2me/syscalls/rimapi/classes/Logger.java
import net.rim.device.api.system.*;

public class Logger
{
	private static long m_guid = 0xba52f65612d86d5aL; //com.linqmap.freemap.Logger
    
    static
    {
        EventLogger.register(m_guid, "FreeMap", EventLogger.VIEWER_STRING);
    }
    
    public static void log(String text)
    {
        System.out.println("FREEMAP LOG: " + text);

        EventLogger.logEvent(m_guid, text.getBytes());
    }
    
    public static void printType(int handle){
      Object obj = CRunTime.getRegisteredObject(handle);
  	  Logger.log("Type is "+ obj);
    }
}

File: WazeWP7/Logger.cs
using System;
using System.IO;
using System.IO.IsolatedStorage;
using System.Windows;

    public class Logger
    {
        public static bool shouldWriteToLog = true;
        public static FileStream logfile = null;
        public static string logs = "";
        public static void log(String text)
        {
            /*
            var store = IsolatedStorageFile.GetUserStoreForApplication();
            if (logfile == null)
            {
                logfile = store.OpenFile("wazelog.txt", FileMode.Create);
            }

            if (shouldWriteToLog)
            {                
                byte[] bytes = Syscalls.StringToAscii(text + "\r\n");
                logfile.Write(bytes, 0, bytes.Length);
                //logfile.Flush();
            }
            else
            {
                logfile.Close();
            }
*/
            if (logs.Length > 300 * 1024)
                logs = "Zeroed";

            logs += "\r\n" + DateTime.Now + ":" + text;
            System.Diagnostics.Debug.WriteLine("FREEMAP LOG: " + text);
        }

        public static void WriteToFile()
        {
            var store = IsolatedStorageFile.GetUserStoreForApplication();
            if (logfile == null)
            {
                logfile = store.OpenFile("wazelog.txt", FileMode.Create);
            }
            byte[] bytes = Syscalls.StringToAscii(logs);
            logfile.Write(bytes, 0, bytes.Length);
            logfile.Close();

            logs = "";
        }

        public static void printType(int handle)
        {
            Object obj = CRunTime.getRegisteredObject(handle);
            Logger.log("Type is " + obj);
        }
    }
